schemas/iso19139/src/main/plugin/iso19139/formatter/jsonld/iso19139-to-jsonld.xsl

```diff
@@ -185,67 +185,45 @@
-        "@id":"<xsl:value-of select="$id"/>",
-        "@type":"Organization"
-        <xsl:for-each select="gmd:organisationName">
-          ,"name": <xsl:apply-templates mode="toJsonLDLocalized"
-                                       select="."/>
-        </xsl:for-each>
-        <xsl:for-each select="gmd:contactInfo/*/gmd:address/*/gmd:electronicMailAddress">
-          ,"email": <xsl:apply-templates mode="toJsonLDLocalized"
-                                       select="."/>
-        </xsl:for-each>
 
-        <!-- TODO: only if children available -->
-        ,"contactPoint": {
-          "@type" : "PostalAddress"
-          <xsl:for-each select="gmd:contactInfo/*/gmd:address/*/gmd:country">
-            ,"addressCountry": <xsl:apply-templates mode="toJsonLDLocalized"
-                                                   select="."/>
-          </xsl:for-each>
-          <xsl:for-each select="gmd:contactInfo/*/gmd:address/*/gmd:city">
-            ,"addressLocality": <xsl:apply-templates mode="toJsonLDLocalized"
-                                                   select="."/>
-          </xsl:for-each>
-          <xsl:for-each select="gmd:contactInfo/*/gmd:address/*/gmd:postalCode">
-            ,"postalCode": <xsl:apply-templates mode="toJsonLDLocalized"
-                                                   select="."/>
-          </xsl:for-each>
-          <xsl:for-each select="gmd:contactInfo/*/gmd:address/*/gmd:deliveryPoint">
-            ,"streetAddress": <xsl:apply-templates mode="toJsonLDLocalized"
-                                                   select="."/>
-          </xsl:for-each>
-          }
-        }
-        <xsl:if test="position() != last()">,</xsl:if>
-      </xsl:for-each>
-    ]
+    "author": [<xsl:for-each select="gmd:identificationInfo/*/gmd:pointOfContact/*[gmd:role/gmd:CI_RoleCode/@codeListValue='author']">
+    <xsl:apply-templates mode="doContact" select="."/>
+    <xsl:if test="position() != last()">,</xsl:if>
+    </xsl:for-each>],
+    "contributor": [<xsl:for-each select="gmd:identificationInfo/*/gmd:pointOfContact/*[gmd:role/gmd:CI_RoleCode/@codeListValue='processor']">
+    <xsl:apply-templates mode="doContact" select="."/>
+    <xsl:if test="position() != last()">,</xsl:if>
+    </xsl:for-each>],
+    "creator": [<xsl:for-each select="gmd:identificationInfo/*/gmd:pointOfContact/*[gmd:role/gmd:CI_RoleCode/@codeListValue='pointOfContact' or gmd:role/gmd:CI_RoleCode/@codeListValue='originator']">
+    <xsl:apply-templates mode="doContact" select="."/>
+    <xsl:if test="position() != last()">,</xsl:if>
+    </xsl:for-each>],
+    "provider" : [<xsl:for-each select="gmd:identificationInfo/*/gmd:pointOfContact/*[gmd:role/gmd:CI_RoleCode/@codeListValue='resourceProvider' or gmd:role/gmd:CI_RoleCode/@codeListValue='custodian']">
+    <xsl:apply-templates mode="doContact" select="."/>
+    </xsl:for-each>
+    <xsl:for-each select="gmd:contact/*">
```

Comment:
This code adds any `gmd:contact` as provider, in the previous version that element was not processed. To confirm that is ok.

Reply:
i see the point, true, not sure if interesting to persist the record properties as properties of the dataset